In [20]:
import pandas as pd

titanic_df = pd.read_csv("titanic-data.csv")

## Basic Information

In [2]:
titatic_df.shape

(891, 12)

In [3]:
titatic_df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='str')

In [4]:
titatic_df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name               str
Sex                str
Age            float64
SibSp            int64
Parch            int64
Ticket             str
Fare           float64
Cabin              str
Embarked           str
dtype: object

In [5]:
titatic_df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [6]:
titatic_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


## Specific Infromation

In [8]:
titatic_df[["Age","Fare"]].mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [9]:
titatic_df[["Age","Fare"]].describe()

,Age,Fare
count,714.000000,891.000000
mean,29.699118,32.204208
std,14.526497,49.693429
min,0.420000,0.000000
25%,20.125000,7.910400
50%,28.000000,14.454200
75%,38.000000,31.000000
max,80.000000,512.329200


### Descriptive Statistics of Age and Fare

The summary statistics provide an overview of the **Age** and **Fare** variables in the Titanic dataset.

| Statistic | Age | Fare |
|----------|---------:|---------:|
| Count | 714 | 891 |
| Mean | 29.70 | 32.20 |
| Standard Deviation | 14.53 | 49.69 |
| Minimum | 0.42 | 0.00 |
| 25th Percentile | 20.13 | 7.91 |
| Median (50%) | 28.00 | 14.45 |
| 75th Percentile | 38.00 | 31.00 |
| Maximum | 80.00 | 512.33 |

**Observation:**

- The **Age** column contains **714** non-null values, indicating that some passenger ages are missing, whereas the **Fare** column has no missing values.
- The average passenger age is **29.7 years**, with most passengers being between **20 and 38 years** old.
- Passenger fares vary considerably, with an average fare of **32.20** and values ranging from **0.00** to **512.33**.
- The high standard deviation of the **Fare** column suggests a wide variation in ticket prices, likely due to differences in passenger class and ticket type.

In [13]:
titatic_df.value_counts("Survived")

Survived
0    549
1    342
Name: count, dtype: int64

In [14]:
titatic_df["Sex"].value_counts()

Sex
male      577
female    314
Name: count, dtype: int64

In [16]:
titatic_df.value_counts(["Sex","Survived"],normalize=True)

Sex     Survived
male    0           0.525253
female  1           0.261504
male    1           0.122334
female  0           0.090909
Name: proportion, dtype: float64

### Distribution of Passengers by Sex and Survival Status

This analysis presents the proportion of passengers based on their gender and survival status.

| Sex | Survived | Proportion |
|------|----------|-----------:|
| Male | No (0) | 52.53% |
| Female | Yes (1) | 26.15% |
| Male | Yes (1) | 12.23% |
| Female | No (0) | 9.09% |

**Observation:**

- The largest group in the dataset consists of **male passengers who did not survive (52.53%)**.
- **Female survivors account for 26.15%** of the total passengers, which is higher than the proportion of male survivors (12.23%).
- Overall, female passengers exhibited a higher survival rate than male passengers, indicating that gender was a significant factor influencing survival.

In [23]:
titanic_df.value_counts(["Sex", "Survived"], normalize=True, ascending=True)

Sex     Survived
female  0           0.090909
male    1           0.122334
female  1           0.261504
male    0           0.525253
Name: proportion, dtype: float64

In [24]:
titanic_df[titanic_df["Sex"]=="female"].value_counts("Survived",normalize=True,ascending=True)

Survived
0    0.257962
1    0.742038
Name: proportion, dtype: float64

### Female Survival Distribution

The analysis shows the distribution of survival outcomes among female passengers. Approximately **74.20%** of female passengers survived, while **25.80%** did not survive. This indicates that the majority of female passengers survived the Titanic disaster, suggesting that gender was an important factor influencing survival.

In [25]:
titanic_df.value_counts("Embarked")

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

In [26]:
titatic_df[titanic_df.Embarked == "S"].value_counts("Sex")

Sex
male      441
female    203
Name: count, dtype: int64

### Distribution of Passengers Embarked from Southampton (S) by Gender

The following analysis shows the number of male and female passengers who boarded the Titanic from **Southampton (S)**.

| Sex | Count |
|------|------:|
| Male | 441 |
| Female | 203 |

**Observation:**

- A total of **644 passengers** embarked from **Southampton (S)**.
- Among them, **441 (68.48%)** were male and **203 (31.52%)** were female.
- Male passengers constituted the majority of passengers embarking from Southampton.

In [27]:
titatic_df[titanic_df.Embarked == "S"].value_counts("Survived")

Survived
0    427
1    217
Name: count, dtype: int64

### Survival Distribution of Passengers Embarked from Southampton (S)

The following analysis shows the survival status of passengers who embarked from **Southampton (S)**.

| Survived | Count |
|----------|------:|
| No (0) | 427 |
| Yes (1) | 217 |

**Observation:**

- A total of **644 passengers** boarded the Titanic from **Southampton (S)**.
- Among them, **427 (66.30%)** did not survive, while **217 (33.70%)** survived.
- The majority of passengers embarking from Southampton did not survive the disaster.